src/components/collapsible_nav/collapsible_nav.tsx

```diff
@@ -160,19 +162,27 @@ export const EuiCollapsibleNav: FunctionComponent<EuiCollapsibleNavProps> = ({
         });
 
   const closeButton = showCloseButton && (
-    <EuiButtonEmpty
-      onClick={collapse}
-      size="xs"
-      iconType="cross"
-      {...closeButtonProps}
-      className={classNames(
-        'euiCollapsibleNav__closeButton',
-        closeButtonProps && closeButtonProps.className
-      )}>
-      <span className="euiCollapsibleNav__closeButtonLabel">
-        <EuiI18n token="euiCollapsibleNav.closeButtonLabel" default="close" />
-      </span>
-    </EuiButtonEmpty>
+    <EuiScreenReaderOnly showOnFocus>
+      <EuiButtonEmpty
+        onClick={collapse}
+        size="xs"
+        textProps={{ className: 'euiCollapsibleNav__closeButtonText' }}
+        iconType="cross"
+        {...closeButtonProps}
+        className={classNames(
+          'euiCollapsibleNav__closeButton',
+          closeButtonProps && closeButtonProps.className
+        )}>
+        <EuiHideFor sizes={['xs']}>
+          <span>
+            <EuiI18n
+              token="euiCollapsibleNav.closeButtonLabel"
+              default="close"
+            />
+          </span>
+        </EuiHideFor>
```

Comment:
My thought is that we could get rid of all these wrappers:
```suggestion
            <EuiI18n
              token="euiCollapsibleNav.closeButtonLabel"
              default="close"
            />
```

And target the new class you created in the textProps to apply the breakpoint styling like...